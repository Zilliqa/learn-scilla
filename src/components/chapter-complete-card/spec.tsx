import React from 'react';
import ReactDOM from 'react-dom';
import * as renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router';
import ChapterCompleteCard from '.';

const t = (s: string) => s;
const navigate = (chapterNum, lessonNum) => console.log('navigate', chapterNum, lessonNum);

describe('Chapter Complete Card tests', () => {
  const baseComponent = (props) => (
    <MemoryRouter initialEntries={['/']}>
      <ChapterCompleteCard navigate={navigate} t={t} total={props.total} chapter={props.chapter} />
    </MemoryRouter>
  );

  describe('basic tests', () => {
    it('matches the snapshot', () => {
      const tree = renderer.create(baseComponent({ total: 2, chapter: 1 })).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('renders without crashing while loading', () => {
      const div = document.createElement('div');
      ReactDOM.render(baseComponent({ total: 2, chapter: 1 }), div);
      ReactDOM.unmountComponentAtNode(div);
    });

    it('renders without crashing after loaded', () => {
      const div = document.createElement('div');
      ReactDOM.render(baseComponent({ total: 2, chapter: 1 }), div);
      ReactDOM.unmountComponentAtNode(div);
    });
  });
});
