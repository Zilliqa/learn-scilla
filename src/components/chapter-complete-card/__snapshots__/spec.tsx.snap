// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chapter Complete Card tests basic tests matches the snapshot 1`] = `
<div
  className="chapter-complete-card card card-body text-center py-5"
  data-test-id="chapter-complete-card"
>
  <div
    className="text-secondary"
  >
    <h3>
      chapter.goodjob
      !
    </h3>
    <p>
      chapter.chapterCompleteMessage
    </p>
    <br />
    <button
      aria-label="next lesson"
      className="btn btn-md type-primary "
      onClick={[Function]}
      type="button"
    >
       
      lesson.nextLesson
       
    </button>
     
    <a
      aria-label="table of contents"
      className="btn type-secondary"
      href="/"
      onClick={[Function]}
    >
      link.tableOfContents
    </a>
  </div>
</div>
`;
